README.md

```diff
@@ -30,6 +30,50 @@ latest JavaScript features; please also see [this wiki page](https://github.com/
 
 + Older browsers: https://mozilla.github.io/pdf.js/legacy/web/viewer.html
 
+## Setup Instructions (Platform-Specific)
+
+### Windows
+1. Install [Node.js](https://nodejs.org/) and [npm](https://www.npmjs.com/get-npm).
+2. Clone the repository and navigate into the project folder:
+   bash
+   git clone https://github.com/mozilla/pdf.js
+   cd pdf.js
+3. Install dependencies
+   npm install
+   npm start
+
+### MacOS/Linux
+1. Install Node.js using Homebrew or your package manager: 
+   brew install node
+2. Clone the repository and navigate into the project folder:
+   git clone https://github.com/your-username/pdf.js
+   cd pdf.js
+3. Install dependencies and start the server:
+   npm install 
+   npm start
+
+If you encounter any issues during setup, please refer to the troubleshooting section
+
+### Trouble Shooting Section
+
+If you encounter issues during setup or development, here are some common problems and their solutions:
+
+ - **Error: 'npm' command not found**: Ensure that Node,js and npm are installed correctly. Verify by running `node -v` and `npm -v`
+ - **Dependency installation fails**: Run `npm cache clean --force` and try `npm install` again. 
+ - **Server not starting**: Check for conflicting processes using the same port (usually 3000). Try running `npx kill-port 300` and restarting the server. 
+
+If your issues persists, feel free to open a discussion or consult the [issues section](https://github.com/mozilla/pdf.js/issues). 
+
+### Contribution Guide (Summary)
```

Comment:
Please note that we want people to read *the entire* https://github.com/mozilla/pdf.js/wiki/Contributing page, to avoid problems, and that one is already linked from the https://github.com/mozilla/pdf.js?tab=readme-ov-file#contributing heading further up.

Hence this section should be removed as far as I'm concerned.